SCClassLibrary/Common/Core/Kernel.sc

```diff
@@ -452,11 +462,28 @@ FunctionDef {
 		_FunctionDefDumpContexts
 		^this.primitiveFailed
 	}
+
+	// These methods are used to indicate that only a subsection of the argName
+	//     and prototypeFrame are used to call the FunctionDef.
+	argNamesForCallStartIndex { ^0 }
+	argNamesForCallSize { ^argNames.size }
+
+	argNamesForCall {
```

Comment:
Actually.... I think I just found a bug in `performWithEnvir` directly related to why the keep is needed....

```supercollider

f = {|a...b| 
	var i = 0; 
	i.debug('i should be 0');
	a.debug('a should be 1');
	b.debug('b should be []');
};

f.(1);
// i should be 0: 0
// a should be 1: 1
// b should be []: []

f.performWithEnvir(\value, (a: 1));
// i should be 0: 0
// a should be 1: 1
// b should be []: [[], 0]    <<<<<<<<< 

```
In side of prototypeframe is also the default values of the variables, and these should not be passed to the performWithList (or any other perform).

This is another reason why I think `makePerformableEnvir` and `makePerformableArray` should live inside `FunctionDef`, because it is complicated and prone to error. I am also going to add a var args argument to these.


For reference.
```supercollider
performWithEnvir { |selector, envir|
		var argNames, args;
		var method = this.class.findRespondingMethodFor(selector);

		if(method.isNil) { ^this.doesNotUnderstand(selector) };

		argNames = method.argNames.drop(1);
		args = method.prototypeFrame.drop(1);
		argNames.do { |name, i|
			var val = envir[name];
			val !? { args[i] = val };
		};

		^this.performList(selector, args)
	}
```

Reply:
> There may be other places in the class library that need to be taken care of once the interface with `defaultArgs` is settled.

I've looked around, in most places there are measures to drop the excess elements, but without comment. So having an explicit `defaultArgs` method will definitely be better.

Reply:
Ah that's a good idea. Alright I'll do that tomorrow.